ethcore/src/blockchain/blockchain.rs

```diff
@@ -383,6 +400,34 @@ impl BlockChain {
 			let best_block_total_difficulty = bc.block_details(&best_block_hash).unwrap().total_difficulty;
 			let best_block_rlp = bc.block(&best_block_hash).unwrap();
 
+			let raw_first = bc.db.get(DB_COL_EXTRA, b"first").unwrap().map_or(Vec::new(), |v| v.to_vec());
+
+			// binary search for the first block.
+			if raw_first.is_empty() {
+				let (mut f, mut hash) = (best_block_number, best_block_hash);
```

Comment:
also, as this doesn't actually change the format of the database it would be strange to build a migration around it.